borg/testsuite/repository.py

```diff
@@ -216,18 +216,18 @@ def segments_in_repository():
         self.repository.commit()
 
         self.repository.append_only = False
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 2
         self.repository.put(b'00000000000000000000000000000000', b'foo')
         self.repository.commit()
         # normal: compact squashes the data together, only one segment
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 4
```

Comment:
- data of the first transaction, 1 segment
- commit, 1 segment
- [compact doesn't do anything here]
- data superseding older data (the put direct above this assert), 1 segment
- commit, 1 segment
- now a total of four segments
- compact copies live data from segment no. 1 (none) to a new segment (so nothing here). Then writes a commit (1 segment), deletes segment no. 1
- total of four segments